### PR TITLE
Fix profile creation and add password to onboarding

### PR DESCRIPTION
**Pull request by chiedoclaude**
## Summary

- **Fix Sentry PERMISSION-SLIP-GO-J**: The `INSERT INTO profiles` query subqueried `auth.users` for the user's email, which failed with `permission denied for schema auth` when `app_backend` lacked auth schema access during the prepared-statement phase. Now passes the email from the JWT session context directly, eliminating the `auth.users` dependency from the profiles INSERT entirely.
- **Add password field to onboarding**: New users are now required to set a password during account creation. The password is set via Supabase's `updateUser` API after profile creation succeeds. If the password set fails, the user still proceeds (non-blocking) and can set a password later.
- **Add shared password validation**: Minimum 8 characters, defined in `shared/validation.json` for frontend/backend parity.

Closes #901

### Developer
- [x] Backend db tests pass (`go test ./db/...`)
- [x] Frontend tests pass (`npm test -- --run`) — all 953 tests green
- [x] TypeScript compiles clean (`npx tsc --noEmit`)
- [x] Go packages compile (`go build ./api/... ./db/...`)

### Manual Testing
- [ ] Sign up as a new user — verify password field appears on onboarding page
- [ ] Complete onboarding with username + password — verify profile is created and password is set
- [ ] Sign out and sign back in using email + password — verify it works
- [ ] Verify the "Create account" button stays disabled until terms are checked AND password meets minimum length

https://claude.ai/code/session_0132mMbDF6z93fjxu1afMrTy

### CODE REVIEW

File: frontend/src/auth/OnboardingPage.tsx [hunk 1]
```diff
@@ -39,6 +41,16 @@ export default function OnboardingPage() {
...
 
+      // Set the user's password via Supabase after profile creation.
+      const { error: pwError } = await supabase.auth.updateUser({
+        password,
+      });
+      if (pwError) {
+        // Profile was created but password failed — log the user in
+        // anyway so they aren't stuck. They can set a password later.
+        console.error("Failed to set password during onboarding:", pwError.message);
+      }
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply frontend/src/auth/OnboardingPage.tsx:57]: Good catch. Updated to show a user-facing warning via setError() so the user knows the password wasn't set and can do it later in Settings. The onboarding still proceeds to the dashboard.

File: db/onboarding_test.go [hunk 2]
```diff
@@ -29,6 +29,9 @@ func TestCreateProfile_Success(t *testing.T) {
 	if profile.CreatedAt.IsZero() {
 		t.Error("expected non-zero created_at")
 	}
+	if profile.Email == nil || *profile.Email != "newuser@example.com" {
+		t.Errorf("expected email %q, got %v", "newuser@example.com", profile.Email)
+	}
 }
```
[the remark this answers is not in the document]
Comment (chiedoclaude) [reply db/onboarding_test.go:35]: Added TestCreateProfile_EmptyEmail — verifies that passing an empty string stores NULL in the email column.

**Comment by chiedoclaude** [comment]
## 🤖 Watch Session Summary

### Changes Made
No changes made during this session.

### Decision Log

#### ✅ Implemented
- greptile-apps[bot] asked for Silent password failure leaves user with no feedback — surface a non-blocking warning → implemented in `00bd3bf`
- greptile-apps[bot] asked for Missing test for empty-email (nil) path — add TestCreateProfile_EmptyEmail → implemented in `00bd3bf`
- greptile-apps[bot] asked for Redundant UserEmail call shadows outer variable — simplify to use existing email var → implemented in `00bd3bf`

---
*Watch session ended after 6 minutes of inactivity. Processed 3 comments across 6 poll cycles.*

**Comment by chiedoclaude** [comment]
⚠️ **Auto-merge failed.** The `--automerge` flag was set, but the merge could not be completed. Please merge manually.
